dipy/denoise/noise_estimate.py

```diff
@@ -88,20 +187,28 @@ def piesno(data, N=1, alpha=0.01, l=100, itermax=100, eps=1e-5, return_mask=Fals
 
     if N in opt_quantile:
         q = opt_quantile[N]
+
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give
+    # up: 
+    if m == 0:
```

Comment:
It' an open suggestion let's say. I was playing with a very noisy, background masked dataset, and it cut 2-3 lines of gm/wm at the top of the coronal slice for this reason I suspect. Relaxing this warning for np.all(data==0) would probably work, since the algorithm already exits if no noise voxel is found , but it's computed if we decide to remove this check + warning.